docs-en/chapter_computational_complexity/iteration_and_recursion.md

```diff
@@ -1,194 +1,199 @@
 
-The number of operations in this summation function is proportional to the size of the input data $n$, or a "linear relationship". In fact, **time complexity describes this "linear relationship"**. This is described in more detail in the next section.
+The number of operations in this summation function is proportional to the size of the input data $n$, or in other words, it has a "linear relationship." In fact, this "linear relationship" is what time complexity describes. This topic will be discussed in more detail in the next section.
 
-### While Loop
+### While Loops
 
-Similar to a `for` loop, a `while` loop is a way to implement iteration. In a `while` loop, the program first checks the condition at each turn, and if the condition is true, it continues, otherwise it ends the loop.
+Similar to `for` loops, `while` loops are another approach for implementing iteration. In a `while` loop, the program checks a condition at the beginning of each iteration; if the condition is true, the execution continues, otherwise, the loop ends.
 
-Below, we use a `while` loop to realize the summation $1 + 2 + \dots + n$ .
+Next, let's use a `while` loop to implement the summation of $1 + 2 + \dots + n$.
 
 ```src
 [file]{iteration}-[class]{}-[func]{while_loop}
 ```
 
-In `while` loops, since the steps of initializing and updating condition variables are independent of the loop structure, **it has more degrees of freedom than `for` loops**.
+**`While` loops provide more flexibility than `for` loops**, especially since they allow for custom initialization and modification of the condition variable at each step.
 
-For example, in the following code, the condition variable $i$ is updated twice per round, which is not convenient to implement with a `for` loop.
+For example, in the following code, the condition variable $i$ is updated twice in each iteration, a scenario that would be less convenient to implement with a `for` loop
 
 ```src
 [file]{iteration}-[class]{}-[func]{while_loop_ii}
 ```
 
-Overall, **`for` loops have more compact code and `while` loops are more flexible**, and both can implement iteration structures. The choice of which one to use should be based on the needs of the particular problem.
+Overall, **`for` loops tend to be more concise, while `while` loops offer more flexibility**. Both can be used to implement iterative structures, and the choice between them should be based on the specific requirements of the problem at hand.
 
 ### Nested Loops
 
-We can nest one loop structure inside another, using the `for` loop as an example:
+Nested loops involve placing one loop structure within another. Taking `for` loops as an example:
 
 ```src
 [file]{iteration}-[class]{}-[func]{nested_for_loop}
 ```
 
-The figure below gives the block diagram of the flow of this nested loop.
+Figure 2-2 provides the flowchart for such a nested loop structure.
 
 ![Block diagram of the flow of nested loops](iteration_and_recursion.assets/nested_iteration.png)
```

Comment:
Update the caption in this line:

![Flowchart of the Nested Loop](iteration_and_recursion.assets/nested_iteration.png)